### ISSUE TITLE
Optimal level of code splitting

### ISSUE THREAD

**Issue by vonwao** [davezuko/react-redux-starter-kit#1052]
Does anyone have any experience or empirical data on the optimal level of code splitting? 

The Fractal Project Structure page seems to suggest a high level of code splitting, splitting each route into a separate bundle.  One of the major advantages of a single-page app is that the experience is very smooth because no server requests are required to load additional pages.  It seems the code-splitting routes negates this advantage, and maybe only reducing your bundle size by 50 or 100 KB.  In a networked world where we are often downloading large files weighing many MBs, and where latency (not throughput) is the biggest issue, I question the logic of doing a lot of code splitting.

**Comment by kolpav** [comment]
Code splitting is not about reducing bundle size but loading only parts of your application necessary for each specific route. Imagine having admin and user section of your application why would you require regular user to load parts of your app which he will never use?

> One of the major advantages of a single-page app is that the experience is very smooth because no server requests are required to load additional pages

It is not very smooth when empty page with tiny login form is 3mb of data. Its just a technique how to negate that heavy initial boot of SPA.

Also I find it much more easier to navigate my code base when I have route specific components, selectors, etc. in its own folder. Even in projects I haven't touched in a few months I can very quickly find my way around because everything is scoped by routes so I just need to focus on one folder at a time and ignore the rest if I want to.

**Comment by mikedevita** [comment]
agreed 👍 having everything under routes/ and fractal sub routes makes it really nice..

**Comment by 6zz** [comment]
same here, I start to like the fractal structure.  I think we can also avoid code splitting by using "component" instead of "getComponent".  I am working on a project where I build the FE with this spa starter kit, but I only need the final generated assets that is served up by php server.  I actually could not get the split code load during runtime because the target url is not relative.  e.g. my /a/b/c/app.js tries to load /0.compx.42345.js instead of /a/b/c/0.compx.42345.js

**Comment by vonwao** [comment]
> Code splitting is not about reducing bundle size

> It is not very smooth when empty page with tiny login form is 3mb of data

You are saying opposite things, your argument is not very congruent.  

Yes, code splitting is good in some cases, to reduce initial page size, and take out things most people will never use (such as admin pages).  But my question is still not answered.

For people who like lots of files and directories and lots of boilerplate, the Fractal Project Structure seems like a good solution.  Also, you can still split your app into multiple directories/modules easily with react-router without all this fractal stuff.
